src/sentry/static/sentry/app/components/confirm.tsx

```diff
@@ -14,268 +14,259 @@ type MessageRenderProps = {
-
   /**
    * Render props to control rendering of the modal in its entirety
    */
   children?:
     | ((renderProps: ChildrenRenderProps) => React.ReactNode)
     | React.ReactElement<{disabled: boolean; onClick: (e: React.MouseEvent) => void}>;
-
   /**
    * Passed to `children` render function
    */
   disabled?: boolean;
-
   /**
    * Callback function when user is in the confirming state
    * called when the confirm modal is opened
    */
   onConfirming?: () => void;
-
   /**
    * User cancels the modal
    */
   onCancel?: () => void;
-
   /**
    * Header of modal
    */
   header?: React.ReactNode;
-} & typeof defaultProps;
-
-type State = {
   /**
-   * Is modal open
+   * Stop event propagation when opening the confirm modal
    */
-  isModalOpen: boolean;
-
+  stopPropagation?: boolean;
   /**
-   * Is confirm button disabled
+   * Disables the confirm button.
+   *
+   * XXX: Once the modal has been opened mutating this property will _not_
+   * propagate into the modal.
+   *
+   * If you need the confirm buttons disabled state to be reactively controlled,
+   * consider using the renderMessage prop, which receives a
+   * `disableConfirmButton` function that you may use to control the state of it.
    */
-  disableConfirmButton: boolean;
+  disableConfirmButton?: boolean;
+  /**
+   * Button priority
+   */
+  priority?: React.ComponentProps<typeof Button>['priority'];
+  /**
+   * Text to show in the cancel button
+   */
+  cancelText?: React.ReactNode;
+  /**
+   * Text to show in the confirmation button
+   */
+  confirmText?: React.ReactNode;
 };
 
-class Confirm extends React.PureComponent<Props, State> {
-  static propTypes = {
-    onConfirm: PropTypes.func.isRequired,
-    confirmText: PropTypes.string.isRequired,
-    cancelText: PropTypes.string.isRequired,
-    priority: PropTypes.oneOf(['primary', 'danger']).isRequired,
-    /**
-     * If true, will skip the confirmation modal and call `onConfirm`
-     */
-    bypass: PropTypes.bool,
-    message: PropTypes.node,
-    /**
-     * Renderer that passes:
-     * `confirm`: Allows renderer to perform confirm action
-     * `close`: Allows renderer to toggle confirm modal
-     */
-    renderMessage: PropTypes.func,
-    disabled: PropTypes.bool,
-    disableConfirmButton: PropTypes.bool,
-    onConfirming: PropTypes.func,
-    onCancel: PropTypes.func,
-    header: PropTypes.node,
-
-    // Stop event propagation when opening the confirm modal
-    stopPropagation: PropTypes.bool,
-  };
-
-  static defaultProps = defaultProps;
-
-  state: State = {
-    isModalOpen: false,
-    disableConfirmButton: this.props.disableConfirmButton,
-  };
-
-  static getDerivedStateFromProps(props: Props, state: State) {
-    // Reset the state to handle prop changes from ConfirmDelete
-    if (props.disableConfirmButton !== state.disableConfirmButton) {
-      return {
-        disableConfirmButton: props.disableConfirmButton,
-      };
+function Confirm({
```

Comment:
I would create another file for ConfirmModal.

Reply:
I would prefer to keep these co-located I think, they are very tightly coupled.